extensions/wikia/ContentReview/tests/ContentReviewHelperTest.php

```diff
@@ -8,61 +8,119 @@ public function setUp() {
-		$helperMock->method( 'isContentReviewTestModeEnabled' )
-			->will( $this->returnValue( $params['isContentReviewTestModeEnabled'] ) );
-		$helperMock->method( 'getRevisionById' )
-			->will( $this->returnValue( $revisionMock ) );
-
-		$helperMock->replaceWithLastApproved( $titleMock, $params['contentType'], $params['text'] );
-
-		$this->assertEquals( $textExpected, $params['text'], $message );
+	public function testReplaceWithLastApprovedRevision( $isPageReviewed, $latestRevID, array $latestReviewedRevision,
+		$isTestModeEnabled, $revisionExists, $originalText, $lastReviewedText, $expectedText, $message
+	) {
+		/**
+		 * Case 1 - if a page does not qualify to be reviewed just return the original text.
+		 */
+		if ( !$isPageReviewed ) {
+			$mockTitle = $this->getMock( '\Title' );
+
+			$mockHelper = $this->getMock( '\Wikia\ContentReview\Helper', [ 'isPageReviewed' ] );
+			$mockHelper->expects( $this->once() )
+				->method( 'isPageReviewed' )
+				->willReturn( $isPageReviewed );
+
+			$textReturn = $mockHelper->replaceWithLastApproved( $mockTitle, 'text/css', $originalText );
+		} else {
+			$mockContentType = 'text/javascript';
+
+			/* @var \Title $titleMock */
+			$titleMock = $this->getMock( '\Title', [ 'getArticleID', 'getLatestRevID' ] );
+			$titleMock->expects( $this->once() )
+				->method( 'getLatestRevID' )
+				->willReturn( $latestRevID );
+
+			/* @var \Wikia\ContentReview\Models\CurrentRevisionModel $currentRevisionModelMock */
+			$currentRevisionModelMock = $this->getMock( 'Wikia\ContentReview\Models\CurrentRevisionModel', [
+				'getLatestReviewedRevision'
+			] );
+			$currentRevisionModelMock->expects( $this->once() )
+				->method( 'getLatestReviewedRevision' )
+				->willReturn( $latestReviewedRevision );
+
+			/* @var \Wikia\ContentReview\Helper $helperMock */
+			$helperMock = $this->getMock( '\Wikia\ContentReview\Helper', [
+				'isPageReviewed',
+				'getCurrentRevisionModel',
+				'isContentReviewTestModeEnabled',
+				'getRevisionById',
+			] );
+			$helperMock->expects( $this->once() )
+				->method( 'isPageReviewed' )
+				->willReturn( $isPageReviewed );
+			$helperMock->expects( $this->once() )
+				->method( 'getCurrentRevisionModel' )
+				->willReturn( $currentRevisionModelMock );
+			$helperMock->expects( $this->any() )
+				->method( 'isContentReviewTestModeEnabled' )
+				->willReturn( $isTestModeEnabled );
+
+			/**
+			 * Handle Case 4 and Case 5 - if the Revision does not exist the value returned
+			 * by getRevisionById is `false` to match the values returned by the original method.
+			 */
+			if ( $revisionExists ) {
+				/* @var \Revision $revisionMock */
+				$revisionMock = $this->getMockBuilder( '\Revision' )
+					->disableOriginalConstructor()
+					->setMethods( [ 'getRawText' ] )
+					->getMock();
+				$revisionMock->expects( $this->any() )
+					->method( 'getRawText' )
+					->willReturn( $lastReviewedText );
+			} else {
+				$revisionMock = false;
```

Comment:
Minor, but to be explicit `Revision::newFromId` returns `null` on a non-existent rev ID, rather than `false`.